xdsl/dialects/experimental/aie.py

```diff
@@ -0,0 +1,1129 @@
+        super().__init__(
+            attributes={"value": value, "action": action, "blocking": blocking},
+            operands=[lock],
+        )
+
+    def print(self, printer: Printer):
+        printer.print("(")
+        printer.print_operand(self.lock)
+        action_str = (
+            '"Acquire"' if self.action.value.data == LOCK_ACQUIRE else '"Release"'
+        )
+        printer.print(", ")
+        printer.print(action_str)
+        printer.print(", ")
+        printer.print(self.value.value.data)
+        printer.print(")")
+
+    @classmethod
+    def parse(cls, parser: Parser) -> UseLockOp:
+        parser.parse_characters("(")
+        lock = parser.parse_operand()
+        parser.parse_characters(",")
+        action = parser.parse_str_literal()
+        print(action, ", ", action == "Acquire")
+        action = (
+            IntegerAttr.from_int_and_width(LOCK_ACQUIRE, 32)
+            if action == "Acquire"
+            else IntegerAttr.from_int_and_width(LOCK_RELEASE, 32)
+        )
+        parser.parse_characters(",")
+        value = IntegerAttr.from_int_and_width(parser.parse_integer(), 32)
+        parser.parse_characters(")")
+
+        blocking = IntegerAttr.from_int_and_width(BLOCKING, 32)
+
+        return UseLockOp(value, action, blocking, lock)
+
+
+@irdl_op_definition
+class WireOp(IRDLOperation):
+    name = "AIE.wire"
+
+    sourceBundle = attr_def(WireBundleAttr)
+    destBundle = attr_def(WireBundleAttr)
+    source = operand_def(IndexType())
+    dest = operand_def(IndexType())
+
+    def __init__(
+        self,
+        sourceBundle: WireBundleAttr,
+        destBundle: WireBundleAttr,
+        source: Operation | SSAValue,
+        dest: Operation | SSAValue,
+    ):
+        super().__init__(
+            attributes={"sourceBundle": sourceBundle, "destBundle": destBundle},
+            operands=[source, dest],
+        )
+
+
+@irdl_op_definition
+class EndOp(IRDLOperation):
+    name = "AIE.end"
+
+    def __init__(self):
+        super().__init__()
+
+
+AIE = Dialect(
+    "aie",
+    [
+        AMSelOp,
+        BufferOp,
+        TileOp,
+        ConnectOp,
+        CoreOp,
+        DMABDOp,
+        DMABDPACKETOp,
+        DMAStartOp,
+        DebugOp,
+        DeviceOp,
+        ExternalBufferOp,
+        FlowOp,
+        GetCascadeOp,
+        GetStreamOp,
+        LockOp,
+        MasterSetOp,
+        MemOp,
+        MemTileDMAOp,
+        NextBDOp,
+        ObjectFifoAcquireOp,
+        ObjectFifoRegisterExternalBuffersOp,
+        ObjectFIFOSubviewAccessOp,
+        createObjectFifo,
+        ObjectFIFOReleaseOp,
+        PLIOOp,
+        PacketDestOp,
+        PacketFlowOp,
+        PacketRuleOp,
+        PacketRulesOp,
+        PacketSourceOp,
+        putCascade,
+        putStream,
+        ShimDMAOp,
+        ShimMuxOp,
+        ShimSwitchBoxOp,
+        SwitchboxOp,
+        UseLockOp,
+        WireOp,
+        EndOp,
+    ],
+    [WireBundleAttr, ObjectFIFO, ObjectFIFOSubview],
```

Comment:
nit:
```suggestion
    [WireBundleAttr, ObjectFIFO, ObjectFIFOSubview,],
```
Trailing commas makes it a little easier on git in the future